### ISSUE TITLE
Stencil start gives stencil-cli out of date error

### ISSUE THREAD

**Issue by AnirudhSingal91** [bigcommerce/stencil-cli#766]
I'm trying to serve my cornerstone theme locally. I have the latest version of stencil-cli installed (v3.5.0). Im getting the error "not ok -- Error: Error: You are using an outdated version of stencil-cli, please run $ npm install -g @bigcommerce/stencil-cli"

ive tried reinstalling stencil-cli but it didnt help. 

### Expected behavior
Expected behaviour is upon running strencil start, my stencil theme should start getting served localy.

### Actual behavior
I get error "not ok -- Error: Error: You are using an outdated version of stencil-cli, please run $ npm install -g @bigcommerce/stencil-cli"

### Steps to reproduce behavior
run stencil start

**Comment by zvuki** [comment]
Please install 3.5.1.

**Comment by AnirudhSingal91** [comment]
> Please install 3.5.1.

it gives the error "not ok -- Error: Could not fetch a list of the store channels: Request failed with status code 404" now

**Comment by bookernath** [comment]
More context here: https://developer.bigcommerce.com/changelog#publications/required-stencil-cli-version-set-to-3-5-0

**Comment by bookernath** [comment]
@AnirudhSingal91 if you're still seeing the issue on `3.5.1`, can you try regenerating your CLI token from the control panel? Sorry for the inconvenience.

**Comment by AnirudhSingal91** [comment]
> @AnirudhSingal91 if you're still seeing the issue on `3.5.1`, can you try regenerating your CLI token from the control panel? Sorry for the inconvenience.
 
I regenerated the token. im still getting the error "not ok -- Error: Could not fetch a list of the store channels: Request failed with status code 404"

**Comment by r742davis** [comment]
@AnirudhSingal91 I'm getting the same exact error on `stencil-cli@3.5.1` after updating and using new tokens:

`not ok -- Error: Could not fetch a list of the store channels: Request failed with status code 403`

**Comment by steve-ross** [comment]
It is silly that you FORCE AN UPDATE every time you release a minor version or point release.  This has killed half my day (and seems to every time you decide to push a new version).

**Comment by zpaquette** [comment]
My team is seeing the same thing as @AnirudhSingal91 and @r724davis

**Comment by steve-ross** [comment]
@zpaquette try updating your API key to include read access on store channels

**Comment by AnirudhSingal91** [comment]
> @zpaquette try updating your API key to include read access on store channels

How does that work? Aren't stencil CLI token and API keys completely independent of each other?

**Comment by rossyoung5** [comment]
> Please install 3.5.1.

To be clear, attempting to resolve this issue led me directly into this concurrent issue: [https://github.com/bigcommerce/stencil-cli/issues/768](https://github.com/bigcommerce/stencil-cli/issues/768)

I had to spend the better part of today trying to figure out what versions of node, npm and stencil-cli work together with no clear guidance from any of the output from stencil-cli. Why was this a forced update, especially with it not working on the latest npm in node 12?

**Comment by MattPua** [comment]
I will mention for myself:

Ran into the `outdated stencil cli` error.

After uninstalling `stencil-cli`, globally and then installing it again globally on version `3.5.1`, it still wasn't working via `npm run start`.

However,

running `stencil init` again and then `stencil start` seems to work fine for now. Not the greatest experience but at least I can get it running for the interim again.

**Comment by drmzio** [comment]
Having the same issue. Was working just fine yesterday. Tried uninstalling the CLI and reinstalling but get an error like `Cannot find module 'mkdirp'` :/

**Comment by HATECORO** [comment]
ye, me too ,fk bc

**Comment by HATECORO** [comment]
I updated BC,then let me update node ,fk ,wst my time .

**Comment by AnirudhSingal91** [comment]
This is still not working and it looks like im going to miss an important deadline because of this issue. I still get the error "not ok -- Error: Could not fetch a list of the store channels: Request failed with status code 404".

Any help will be appreciated!

**Comment by rossyoung5** [comment]
@AnirudhSingal91 Have you tried @steve-ross's workaround in [#768](https://github.com/bigcommerce/stencil-cli/issues/768) ?

> #### Workaround for NPM 7.x
> ```
> npm i -g npm@6
> npm i -g @bigcommerce/stencil-cli
> npm i -g npm@latest
> ```

**Comment by AnirudhSingal91** [comment]
@rossyoung5 Yes. didnt work.

**Comment by steve-ross** [comment]
> @AnirudhSingal91 Have you tried @steve-ross's workaround in [#768](https://github.com/bigcommerce/stencil-cli/issues/768) ?
> 
> > #### Workaround for NPM 7.x
> > ```
> > npm i -g npm@6
> > npm i -g @bigcommerce/stencil-cli
> > npm i -g npm@latest
> > ```
I opened this one ;)

**Comment by AnirudhSingal91** [comment]
@steve-ross Yes did so. No help. I end up getting the same error code as before.

**Comment by steve-ross** [comment]
@AnirudhSingal91 you can try doing a cache clean after you downgrade to npm@6 with npm cache clean --force

**Comment by AnirudhSingal91** [comment]
@steve-ross 
Windows 10
node 12.0.0
npm 6.9.0
stencil-cli 3.5.1

**Comment by steve-ross** [comment]
You may also want to reinit stencil... I had to do that even though my project was on 3.5.0 OR ensure you have an apiHost in your `config.stencil.json`

ex: 
```json
{
  "customLayouts": {
    "brand": {},
    "category": {},
    "page": {},
    "product": {}
  },
  "apiHost": "https://api.bigcommerce.com",
  "normalStoreUrl": "https://somewhere.mybigcommerce.com/",
  "port": 3000
}
```

**Comment by steve-ross** [comment]
@r742davis I was just typing something like this up... BigC shouldn't introduce breaking changes in point releases. DEFAULT YOUR DAMN VARIABLES!

**Comment by AnirudhSingal91** [comment]
omg @steve-ross @r742davis!! it worked. idk what worked exactly. but it worked. Most probably it was the apiHost that was getting set incorrectly.

**Comment by steve-ross** [comment]
yeah, I think is a bug in stencil-cli that is defaulting the apiHost variable to the store url instead of the actual api endpoint

**Comment by AnirudhSingal91** [comment]
@steve-ross ive been having another issue with stencil bundle command. i get the error "Error: Cannot find module 'babel-plugin-dynamic-import-webpack'". do you have any idea about it?

**Comment by steve-ross** [comment]
nope but it looks like your project is missing that package... `npm i -D babel-plugin-dynamic-import-webpack` then retry

**Comment by AnirudhSingal91** [comment]
@steve-ross 
I tried that.
It leads me to Error: Cannot find module 'babel-plugin-transform-regenerator'

when i do npm i -D babel-plugin-transform-regenerator, it leads me to 

> Error: Cannot find module 'babel-preset-env'

when i do npm i -D babel-preset-env, it leads me to 

> ERROR in ./assets/js/app.js
> Module build failed (from ./node_modules/babel-loader/lib/index.js):

**Comment by AnirudhSingal91** [comment]
@ do i need to have the latest version of npm installed?

**Comment by steve-ross** [comment]
Likely you're on an old theme that didn't have all the devtools specified in the package.json. I'd take a look at the devDependencies in Cornerstone and see if you can get it working that way. https://github.com/bigcommerce/cornerstone/blob/master/package.json#L30

When things go haywire blow away `./node_modules` and `npm install` again

**Comment by AnirudhSingal91** [comment]
my package.json is excatly the same. though my root directory doesnt have babel.config.js file. Could that be causing this?

**Comment by AnirudhSingal91** [comment]
@steve-ross you were right. there was some issue with my theme files. Got the latest theme and its working perfectly

**Comment by kalwiggins** [comment]
my stencil local does prop up after updating the cli, but it takes 2 or 3 minutes to do it now, as well as to refresh it or push. and I did rerun stencil init
- node 12.13.0 on an M1
